tests/e2e_test.go

```diff
@@ -0,0 +1,444 @@
+				AllowedRelayerAddresses: []common.Address{},
+				Message:                 []byte{1, 2, 3, 4},
+			},
+		)
+		Expect(err).Should(BeNil())
+
+		// Send a transaction to the Teleporter contract
+		tx := newTestTeleporterTransaction(chainAIDInt, teleporterContractAddress, nonceA, data)
+
+		txSigner := types.LatestSignerForChainID(chainAIDInt)
+		signedTx, err := types.SignTx(tx, txSigner, fundedKey)
+		Expect(err).Should(BeNil())
+
+		subA, err := chainAWSClient.SubscribeNewHead(ctx, newHeadsA)
+		Expect(err).Should(BeNil())
+		defer subA.Unsubscribe()
+
+		log.Info("Sending Teleporter transaction on source chain", "destinationChainID", blockchainIDB, "txHash", signedTx.Hash())
+		err = chainARPCClient.SendTransaction(ctx, signedTx)
+		Expect(err).Should(BeNil())
+
+		// Sleep to ensure the new block is published to the subscriber
+		time.Sleep(5 * time.Second)
+		receipt, err := chainARPCClient.TransactionReceipt(ctx, signedTx.Hash())
+		Expect(err).Should(BeNil())
+		Expect(receipt.Status).Should(Equal(types.ReceiptStatusSuccessful))
+	})
+
+	ginkgo.It("Relay message to destination", ginkgo.Label("Teleporter", "RelayMessage"), func() {
+		ctx := context.Background()
+
+		// Get the latest block from Subnet A, and retrieve the warp message from the logs
+		log.Info("Waiting for new block confirmation")
+		newHeadA := <-newHeadsA
+		blockHashA := newHeadA.Hash()
+
+		log.Info("Fetching relevant warp logs from the newly produced block")
+		logs, err := chainARPCClient.FilterLogs(ctx, interfaces.FilterQuery{
+			BlockHash: &blockHashA,
+			Addresses: []common.Address{warp.Module.Address},
+		})
+		Expect(err).Should(BeNil())
+		Expect(len(logs)).Should(Equal(1))
+
+		// Check for relevant warp log from subscription and ensure that it matches
+		// the log extracted from the last block.
+		txLog := logs[0]
+		log.Info("Parsing logData as unsigned warp message")
+		unsignedMsg, err := avalancheWarp.ParseUnsignedMessage(txLog.Data)
+		Expect(err).Should(BeNil())
+
+		// Set local variables for the duration of the test
+		unsignedWarpMessageID := unsignedMsg.ID()
+		unsignedWarpMsg := unsignedMsg
+		log.Info("Parsed unsignedWarpMsg", "unsignedWarpMessageID", unsignedWarpMessageID, "unsignedWarpMessage", unsignedWarpMsg)
+
+		// Loop over each client on chain A to ensure they all have time to accept the block.
+		// Note: if we did not confirm this here, the next stage could be racy since it assumes every node
+		// has accepted the block.
+		waitForAllValidatorsToAcceptBlock(ctx, chainANodeURIs, blockchainIDA, newHeadA.Number.Uint64())
+
+		// Get the aggregate signature for the Warp message
+		log.Info("Fetching aggregate signature from the source chain validators")
+		warpClient, err := warpBackend.NewWarpClient(chainANodeURIs[0], blockchainIDA.String())
+		Expect(err).Should(BeNil())
+		signedWarpMessageBytes, err := warpClient.GetAggregateSignature(ctx, unsignedWarpMessageID, params.WarpQuorumDenominator)
```

Comment:
do you know the pros/cons of using `getAggregateSignature` vs how we get signatures and aggregate ourselves in the relayer?

Reply:
Here, it's the simplest way to get the aggregate signature in the end to end test.

This is outside the scope of this PR, but an off chain relayer aggregating signatures may wish to do so by requesting individual signatures from validators directly because:
- Signature aggregation wouldn't depend on a single node's availability
- At least one fewer network round trips since when compared to requesting an external node collect and aggregate signatures

In practice, the first point is probably the more relevant of the two.

Reply:
Yep, agree here. In practice, the single node may already have a complete aggregate signature that it can return directly rather than needing to query its peers (for instance, if it caches the results), so that could cut down on the total amount of network requests.

Outside the scope either way here though 👍